packages/plugin-core/src/commands/Goto.ts

```diff
@@ -49,22 +63,36 @@ export class GotoCommand extends BasicCommand<CommandOpts, CommandOutput> {
     return { kind };
   }
 
-  async gatherInputs(): Promise<CommandInput | undefined> {
-    return {};
-  }
   async execute(): Promise<CommandOutput> {
-    const { vaults, engine } = ExtensionProvider.getDWorkspace();
+    const externalLink = getURLAt(VSCodeUtils.getActiveTextEditor());
+    const noteLink = await getLinkFromSelectionWithWorkspace();
 
-    const link = await getLinkFromSelectionWithWorkspace();
-    if (!link) {
-      return {
-        error: new DendronError({ message: "selection is not a valid link" }),
-      };
+    /* If the link read is not a valid link, exit from the command with a message */
+    if (!externalLink && !noteLink) {
+      const error = DendronError.createFromStatus({
+        status: ERROR_STATUS.INVALID_STATE,
+        message: `no valid path or URL selected`,
+      });
+      this.L.error({ error });
+      return { error };
+    }
+
+    /* Depending on the link type selected, execute different command logic */
+    if (noteLink) {
+      return this.goToNoteLink(noteLink);
+    } else {
+      return this.goToExternalLink(externalLink);
     }
+  }
+
+  private async goToNoteLink(
+    noteLink: Awaited<ReturnType<typeof getLinkFromSelectionWithWorkspace>>
```

Comment:
`Awaited` needs to be imported

Reply:
I couldn't import `Awaited` from typescript, [so I typed `any` instead](https://github.com/dendronhq/dendron/pull/3175/commits/1067b5b939277871d0529de9a136ea3adb4c5ffc)